next/components/forms/segments/MigrationForm/MigrationForm.tsx

```diff
@@ -0,0 +1,109 @@
+import { formatUnicorn } from '@utils/string'
+import { AccountError } from '@utils/useAccount'
+import useHookForm from '@utils/useHookForm'
+import Alert from 'components/forms/info-components/Alert'
+import AccountLink from 'components/forms/segments/AccountLink/AccountLink'
+import LoginAccountLink from 'components/forms/segments/LoginAccountLink/LoginAccountLink'
+import Button from 'components/forms/simple-components/Button'
+import InputField from 'components/forms/widget-components/InputField/InputField'
+import { useTranslation } from 'next-i18next'
+import { useRouter } from 'next/router'
+import { Controller } from 'react-hook-form'
+
+interface Data {
+  email: string
+}
+
+interface Props {
+  onSubmit: (email: string) => Promise<any>
+  error?: AccountError | null | undefined
+}
+
+const MigrationForm = ({ onSubmit, error }: Props) => {
+  const router = useRouter()
+  const email =
+    router.query.email && typeof router.query.email === 'string'
+      ? decodeURIComponent(router.query.email)
+      : ''
+
+  const fullname =
+    router.query.fullname && typeof router.query.fullname === 'string'
+      ? decodeURIComponent(router.query.fullname)
+      : ''
+
+  const schema = {
+    type: 'object',
+    properties: {
+      email: {
+        type: 'string',
+        minLength: 1,
+        // validate input, not query param
+        format: email ? undefined : 'email',
```

Comment:
to make this slightly less confusing, I propose renaming `email` -> `queryEmail` and `fullname` -> `queryFullname`